### PR TITLE
Migrating from boostTest to googleTest.

### PR DESCRIPTION
**Pull request by FeniksFire**
Because _googletest_ has a great integration with _googlemock_ that allows create mocks and it's more popular and faster than _boosttest_ I decided to migrate our tests.

For now should be ok but I will check it again later and fix some things like: improve formatting, delete our logger from tests and etc.
If anyone was looking for the necessary references then all can be found in this [docs](https://github.com/google/googletest/blob/master/googletest/docs/Documentation.md).

### CODE REVIEW

File: test/map/MapComparer.cpp [hunk 1]
```diff
@@ -251,25 +249,24 @@ void MapComparer::compareTerrain()
...
 		}
 }
 
 void MapComparer::compare()
 {
-	BOOST_REQUIRE_NE((void *) actual, (void *) expected); //should not point to the same object
-	BOOST_REQUIRE_MESSAGE(actual != nullptr, "Actual map is not defined");
-	BOOST_REQUIRE_MESSAGE(expected != nullptr, "Expected map is not defined");
+	ASSERT_NE((void *) actual, (void *) expected); //should not point to the same object
+	EXPECT_TRUE(actual != nullptr) << "Actual map is not defined";
```
Comment (alexvins) [inline test/map/MapComparer.cpp]: Assertion required

File: test/map/MapComparer.cpp [hunk 2]
```diff
@@ -327,17 +319,17 @@ bool JsonMapComparer::isEmpty(const JsonNode & value)
...
+		EXPECT_TRUE(false) << "Unknown Json type";
 		return false;
 	}
 }
 
 void JsonMapComparer::check(const bool condition, const std::string & message)
 {
 	if(strict)
-		BOOST_REQUIRE_MESSAGE(condition, buildMessage(message));
+		EXPECT_TRUE(condition) << buildMessage(message);
```
Comment (alexvins) [inline test/map/MapComparer.cpp]: Assertion required

**Comment by alexvins** [comment]
Also as long as main reason of this refactoring is googlemock support, before this branch will be merged it should include at least one new working test with mocks usage.

**Comment by ArseniyShestakov** [comment]
I sorry that I forgot to tell you that when you asked, but you need to remove "-DENABLE_TEST=0" from .travis.yml so test building is enabled for Travis targets. AppVeyor build with tests passed fine and everything work on my local PC with GCC 6.3.

File: test/battle/CHealthTest.cpp [hunk 3]
```diff
@@ -9,67 +9,58 @@
...
-		baseAmount(UNIT_AMOUNT),
-		health(this)
-	{
-		health.init();
-	}
-
-	int32_t maxHealth;
-	int32_t baseAmount;
+	UnitHealthInfoMock mock;
+	HealthTest() : health(&mock){
```
Comment (alexvins) [inline test/battle/CHealthTest.cpp]: Formatting

File: test/battle/CHealthTest.cpp [hunk 4]
```diff
@@ -86,155 +77,167 @@ static void checkHeal(CHealth & health, EHealLevel level, EHealPower power, cons
...
+TEST_F(HealthTest, singleUnitStack)
 {
 	//related to issue 2612
 
 	//one Titan
-	baseAmount = 1;
-	maxHealth = 300;
+
+	EXPECT_CALL(mock, unitMaxHealth()).WillRepeatedly(Return(1));
+	EXPECT_CALL(mock, unitBaseAmount()).WillRepeatedly(Return(300));
```
Comment (alexvins) [inline test/battle/CHealthTest.cpp]: unitMaxHealth should be 300 and unitBaseAmount should be 1.
You can leave unitMaxHealth = 1, unitBaseAmount = 300 as separate test case, it also should pass.

**Comment by alexvins** [comment]
@FeniksFire is this PR done or still in progress?

**Comment by FeniksFire** [comment]
@alexvins well generally "migration" can be considered as finish, still want make other fixes like creating mock for VLC handlers (to remove dependencies in tests from the global class) but I can do it in other pr and there isn't sense for others to wait.
So, I will change the title, and you can merge it if you want.

**Pull request merged**